src/foam/nanos/crunch/UserCapabilityJunctionDAO.js

```diff
@@ -97,18 +101,106 @@ foam.CLASS({
 
       return getDelegate().put_(x, obj);
       
       `
     },
+    {
+      name: 'saveDataToDAO',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'obj',
+          type: 'foam.nanos.crunch.UserCapabilityJunction'
+        }
+      ],
+      documentation: `
+      We may or may not want to store the data in its own dao, based on the nature of the data. 
+      For example, if the data for some UserCapabilityJunction is a businessOnboarding object, we may want to store this object in 
+      the businessOnboardingDAO for easier access.
+      If the data on an UserCapabilityJunction should be stored in some DAO, the daoKey should be provided on its corresponding Capability object.
+      `,
+      javaCode: `
+      DAO capabilityDAO = (DAO) x.get("capabilityDAO");
+      Capability capability = (Capability) capabilityDAO.find(obj.getTargetId());
+      
+      String daoKey = capability.getDaoKey();
+      if ( daoKey == null ) return;
+      
+      DAO dao = (DAO) x.get(daoKey);
+      if ( dao == null ) return;
+
+      if ( dao.getOf().getId() == (obj.getData()).getClassInfo().getId() ) {
+        try {
+          dao.put(obj.getData());
+        } catch (Exception e) {
+          Logger logger = (Logger) x.get("logger");
+          logger.debug("Data cannot be added to " + daoKey + " for UserCapabilityJunction object : " + obj.getId() );
+        }
+      }
+      `
+    },
+    {
+      name: 'configureJunctionExpiry',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'obj',
+          type: 'foam.nanos.crunch.UserCapabilityJunction'
+        },
+        {
+          name: 'old',
+          type: 'foam.nanos.crunch.UserCapabilityJunction'
+        }
+      ],
+      type: 'foam.core.FObject',
+      documentation: `Set the expiry of a userCapabilityJunction based on the duration or expiry set on the capability, which
+      ever one comes first`,
+      javaCode: `
+      // Only update the expiry for non-active junctions, i.e., non-expired, non-pending, or granted junctions whose expiry is not yet set
+      if ( ( old != null && old.getStatus() == CapabilityJunctionStatus.GRANTED && old.getExpiry() != null ) || obj.getStatus() != CapabilityJunctionStatus.GRANTED ) 
+        return obj;
+
+      DAO capabilityDAO = (DAO) x.get("capabilityDAO");
+      Capability capability = (Capability) capabilityDAO.find((String) obj.getTargetId());
+      Date junctionExpiry = capability.getExpiry();
+      
+      if ( capability.getDuration() > 0 ) {
+        DateFormat format = new SimpleDateFormat("yyyy/MM/dd");
+        Date today = new Date();
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(today);
+        calendar.add(Calendar.DATE, capability.getDuration());
+
+        if ( junctionExpiry == null ) junctionExpiry = calendar.getTime();
```

Comment:
Please don't ever format if/else's this way. If you have an else part then indent both parts and use {}'s.